mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -373,11 +385,57 @@ static bool isUnfusedKernelStore(ThreadwiseWriteAllOp store) {
   return ret;
 }
 
+static SmallVector<Attribute> extractVector(ArrayAttr arrayAttr) {
+  return llvm::to_vector<4>(arrayAttr.getAsRange<Attribute>());
+}
+
+LogicalResult
+ReduceRewritePattern::matchAndRewrite(rock::ReduceOp reduceOp,
+                                      PatternRewriter &rewriter) const {
+  Location loc = reduceOp.getLoc();
+  ThreadwiseWriteAllOp threadwiseWriteOp =
+      traceToThreadwiseWrite(reduceOp.getIn());
+  if (reduceOp.getReduceMethod() != ReduceMethod::Sum) {
+    return reduceOp.emitError("We only support sum reductions.!");
+  }
+  if (!threadwiseWriteOp) {
+    LLVM_DEBUG(llvm::dbgs() << "rock.reduce input is not leading to "
+                               "ThreadwiseWriteAllOp of the previous op.");
+    return failure();
+  }
+  int64_t reductionAxis = reduceOp.getAxisAttr().getInt();
```

Comment:
So as to avoid fusing multiple reductions, check to make sure the store method is either `set` or `atomic_add` up here.

Reply:
done.!